src/util/key.rs

```diff
@@ -109,6 +109,21 @@ impl PublicKey {
         }
     }
 
+    /// Read the public key from a reader
+    pub fn read_from<R: io::Read>(mut reader: R) -> Result<Self, io::Error> {
+        let mut bytes = [0; 65];
+        let byte_sl;
+        reader.read_exact(&mut bytes[0..1])?;
+        if bytes[0] < 4 {
+            reader.read_exact(&mut bytes[1..33])?;
+            byte_sl = &bytes[0..33];
+        } else {
+            reader.read_exact(&mut bytes[1..65])?;
+            byte_sl = &bytes[0..65];
+        }
+	Self::from_slice(byte_sl).map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))
```

Comment:
indentation

Reply:
fixed. I need to find whatever broken vim setting is making it do that..